src/commands/key2ds.rs

```diff
@@ -0,0 +1,169 @@
+    #[arg()]
+    keyfile: PathBuf,
+}
+
+pub fn parse_digest_alg(arg: &str) -> Result<DigestAlg, Error> {
+    if let Ok(num) = arg.parse() {
+        let alg = DigestAlg::from_int(num);
+        if alg.to_mnemonic().is_some() {
+            Ok(alg)
+        } else {
+            Err(Error::from("unknown algorithm number"))
+        }
+    } else {
+        DigestAlg::from_mnemonic(arg.as_bytes())
+            .ok_or(Error::from("unknown algorithm mnemonic"))
+    }
+}
+
+impl Key2ds {
+    pub fn execute(self) -> Result<(), Error> {
+        let mut file = std::fs::File::open(&self.keyfile).map_err(|e| {
+            format!(
+                "Failed to open public key file \"{}\": {e}",
+                self.keyfile.display()
+            )
+        })?;
+        let zonefile =
+            domain::zonefile::inplace::Zonefile::load(&mut file).unwrap();
+        for entry in zonefile {
+            let entry = entry.map_err(|e| {
+                format!(
+                    "Error while reading public key from file \"{}\": {e}",
+                    self.keyfile.display()
+                )
+            })?;
+
+            // We only care about records in a zonefile
+            let Entry::Record(record) = entry else {
+                continue;
+            };
+
+            let class = record.class();
+            let ttl = record.ttl();
+            let owner = record.owner();
+
+            // Of the records that we see, we only care about DNSKEY records
+            let ScannedRecordData::Dnskey(dnskey) = record.data() else {
+                continue;
+            };
+
+            // if ignore_sep is specified, we accept any key
+            // otherwise, we only want SEP keys
+            if !self.ignore_sep && !dnskey.is_secure_entry_point() {
+                continue;
+            }
+
+            let key_tag = dnskey.key_tag();
+            let sec_alg = dnskey.algorithm();
+            let digest_alg = self.determine_hash(sec_alg);
+
+            if digest_alg == DigestAlg::GOST {
+                return Err("Error: the GOST algorithm is deprecated and must not be used. Try a different algorithm.".into());
+            }
+
+            let digest = dnskey
+                .digest(&owner, digest_alg)
+                .map_err(|e| format!("Error computing digest: {e}"))?;
+
+            let ds = Ds::new(key_tag, sec_alg, digest_alg, digest).expect(
+                "Infallible because the digest won't be too long since it's a valid digest",
+            );
+
+            let rr = Record::new(owner, class, ttl, ds);
+
+            if self.write_to_stdout {
+                println!("{}", rr.display_zonefile(false));
+            } else {
+                let owner = owner.fmt_with_dot();
+                let sec_alg = sec_alg.to_int();
+                let filename =
+                    format!("K{owner}+{sec_alg:03}+{key_tag:05}.ds");
+                let mut out_file = File::create(&filename).map_err(|e| {
```

Comment:
Would it make sense to use `File::create_new()` here?

Reply:
That is, if we have consensus that we want to break compat here, I'd be happy to change it.

Reply:
Now that we have different argument parsing for ldns and dnst, and we can use different defaults, we can probably also introduce differences here. Maybe a simple `running_as_ldns: bool` (clap-hidden) member is enough or should it be an enum like `MultiCallMode::{LDNS,DNST}`/`CompatMode::{None/DNST,LDNS}` (or similar)?

Reply:
> Now that we have different argument parsing for ldns and dnst, and we can use different defaults, we can probably also introduce differences here. Maybe a simple `running_as_ldns: bool` (clap-hidden) member is enough or should it be an enum like `MultiCallMode::{LDNS,DNST}`/`CompatMode::{None/DNST,LDNS}` (or similar)?

Oh you implemented that already with the force_overwrite

Reply:
Yep!